Lesson05/script.js

```diff
@@ -0,0 +1,50 @@
+// prompt tells use to enter Q (quit), W (withdraw), D (deposit), B (balance), Q (quit)
+
+// if enter Q program quits
+
+// if enter W user is propted to enter an amount to withdraw - after entering amount should be
+// able to type another option
+
+// if enter D user is propted to enter an amount to deposit - after entering amount should be
+// able to type another option
+
+// if enter B user will see their balance - after viewing balance shold be able to type another option
+
+// program will loop asking for input until user enters Q to quit
+
+
+let balance = 1000;
+
+function bankApp(actions) {
+  const input = prompt(actions);
+
+  switch (input) {
+    case 'Q':
+      alert('Thanks for banking with us!');
+      break;
+    case 'W':
+      let withdrawNum = prompt('Enter an amount to withdraw:');
+      balance -= Number(withdrawNum);
+
+      alert('Current balance: ' + '$' + balance);
+      do {
+        bankApp(actions);
+      } while (input = 'W' || 'D' || 'B');
+      break;
+    case 'D':
+      let depositNum = prompt('Enter an amount to deposit:');
+      balance += Number(depositNum);
+
+      alert('Current balance: ' + '$' + balance);
+      do {
+        bankApp(actions);
```

Comment:
recursion? Maybe a bit unneeded for this exercise

Reply:
I didn't even know that's what I was doing- haha!  I appreciated seeing your solution to this problem in class.  It was super helpful and I'm feeling better about how to structure my javascript more efficiently.

Reply:
Yeah, a function that calls itself from inside. THE CALL IS COMING FROM INSIDE THE HOUSE!!!  :)